lib/pipenv.js

```diff
@@ -28,31 +65,48 @@ async function pipfileToRequirements() {
   }
 
   try {
-    try {
-      await spawn('pipenv', ['lock', '--keep-outdated'], {
-        cwd: this.servicePath,
-      });
-    } catch (e) {
-      const stderrBufferContent =
-        (e.stderrBuffer && e.stderrBuffer.toString()) || '';
+    // Get and validate pipenv version
+    if (this.log) {
+      this.log.info('Getting pipenv version');
+    } else {
+      this.serverless.cli.log('Getting pipenv version');
+    }
+
+    const pipenvVersion = await getPipenvVersion();
+    let res;
 
-      if (stderrBufferContent.includes('must exist to use')) {
-        // No previous Pipfile.lock, we will try to generate it here
-        await spawn('pipenv', ['lock'], {
+    if (semver.gt(pipenvVersion, LEGACY_PIPENV_VERSION)) {
```

Comment:
Is that actually going to work? It doesn't seem like `pipenv` is using proper semver versioning

Reply:
Yes, see the additional passing test runner matrix values running both pipenv versions: https://github.com/serverless/serverless-python-requirements/pull/742/files#diff-ed51950595c5b713cdfd2437e37b2ff83a366772307810e1cf5d2226f096d082R19

While the `pipenv` versions don't seem like traditional semver versions, they appear at least semver compliant, e.g. `2022.8.5` < `2022.8.13`.

Reply:
thanks for clarification 🙇